test/end-to-end/signin.test.js

```diff
@@ -669,6 +670,138 @@ describe("SuperTokens SignIn", function () {
+        });
+
+        it("Should have default values in the signin request payload", async function () {
+            await page.evaluate(() => window.localStorage.setItem("SHOW_SIGNIN_DEFAULT_FIELDS", "YES"));
+
+            await page.reload({
+                waitUntil: "domcontentloaded",
+            });
+
+            const expectedDefautlValues = [
+                { id: "email", value: "abc@xyz.com" },
+                { id: "password", value: "fakepassword123" },
+            ];
+
+            let assertionError = null;
+            let interceptionPassed = false;
+
+            const requestHandler = async (request) => {
+                if (request.url().includes(SIGN_IN_API) && request.method() === "POST") {
+                    try {
+                        const postData = JSON.parse(request.postData());
+                        expectedDefautlValues.forEach(({ id, value }) => {
+                            let findFormData = postData.formFields.find((inputData) => inputData.id === id);
+                            if (findFormData) {
+                                assert.strictEqual(findFormData["value"], value, `Mismatch in payload for key: ${id}`);
+                            } else {
+                                throw new Error("Field not found in req.data");
+                            }
+                        });
+                        interceptionPassed = true;
+                        return request.respond({
+                            status: 200,
+                            headers: {
+                                "access-control-allow-origin": TEST_CLIENT_BASE_URL,
+                                "access-control-allow-credentials": "true",
+                            },
+                            body: JSON.stringify({
+                                status: "OK",
+                            }),
+                        });
+                    } catch (error) {
+                        assertionError = error; // Store the error
+                    }
+                }
+                return request.continue();
+            };
+
+            await page.setRequestInterception(true);
+            page.on("request", requestHandler);
+
+            try {
+                // Perform the button click and wait for all network activity to finish
+                await Promise.all([page.waitForNetworkIdle(), submitForm(page)]);
+            } finally {
+                page.off("request", requestHandler);
+                await page.setRequestInterception(false);
+            }
+
+            if (assertionError) {
+                throw assertionError;
+            }
+
+            if (!interceptionPassed) {
+                throw new Error("test failed");
+            }
+        });
+    });
+
+    describe("Check if nonOptionalErrorMsg works as expected", function () {
+        it("Check on blank form submit nonOptionalErrorMsg gets displayed as expected", async function () {
```

Comment:
Not really an issue but these are worded differently from the above.
```suggestion
    describe("nonOptionalErrorMsg", function () {
        it("should be displayed on a blank form submit", async function () {
```